Cargo.toml

```diff
@@ -3,23 +3,21 @@ name = "trin"
 version = "0.1.0"
 authors = ["Jacob Kaufmann <jacobkaufmann18@gmail.com>", "Jason Carver <ut96caarrs@snkmail.com>"]
 edition = "2018"
+default-run = "trin"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+log = "0.4.14"
 tokio = { version = "1.8.0", features = ["full"] }
+tracing = "0.1.26"
+tracing-subscriber = "0.2.18"
+trin-core = { path = "trin-core" }
 trin-history = { path = "trin-history" }
 trin-state = { path = "trin-state" }
 
 [workspace]
 members = [
-    "trin-state",
     "trin-history",
-    "trin-core"
-]
-
-default-members = [
     "trin-state",
-    "trin-history",
-    "trin-core"
 ]
```

Comment:
What is the motivation for making `trin-core` a dependency and `trin-history` and `trin-state` as members of workspace? Are `trin-core`, `trin-history`, and `trin-state` packages or crates, making the entire directory as a workspace?

Reply:
Overall my goal was to allow the CLI API written out in this pr's description. But, it's possible that I didn't choose the best configuration setup here. There's no need to expose `trin-core` to the user, since it's all internal code. But, I wanted to expose the ability for users to run `trin-history` and `trin-state` packages w/o the common `trin` orchestrator logic, if that's what they want to do for whatever reason. Did you see something I missed in this current setup?

Reply:
Haha no, that makes total sense. I am just confused on all Cargo's terminology on crates, workspaces, packages, but what you mentioned helps makes things clearer for me.